<a href="https://colab.research.google.com/github/Sakith-N/Statistical-Learning-e22252/blob/assignment-2/Q4.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

1. Ω = [0,1]
2. X is ameasurable function X:Ω → [0,1] such that for any outcome  ω, X(ω)=ω
3. $f(x) = \begin{cases} \frac{1}{B(2,18)} x^{2-1} (1-x)^{18-1} = \frac{1}{B(2,18)} x(1-x)^{17} & 0 < x < 1 \\ 0 & \text{otherwise} \end{cases}$
4. for the function to be valid,

*   Non-negativity: For $x \in (0, 1)$, both $x$ and $(1-x)$ are positive, so $f(x) \ge 0$
*   Integration to Unity: By definition of the Beta function, $\int_0^1 x^{\alpha-1}(1-x)^{\beta-1} \, dx = B(\alpha, \beta)$. Therefore, $\int_0^1 \frac{1}{B(2,18)} x(1-x)^{17} \, dx = \frac{B(2,18)}{B(2,18)} = 1$.



In [7]:
from scipy.stats import beta

alpha = 2
beta_p = 18

# 5
Ex = beta.mean(alpha,beta_p)

# 6
var = beta.var(alpha,beta_p)

# 7
p_0_1 = beta.cdf(0.10,alpha,beta_p)

# 8
p_0_05to0_15 = beta.cdf(0.15,alpha,beta_p) - beta.cdf(0.05,alpha,beta_p)

print(f"5. Expectation E[X]: {Ex:.4f}")
print(f"6. Variance Var(X):  {var:.6f}")
print(f"7. P(X <= 0.10):     {p_0_1:.4f}")
print(f"8. P(0.05 <= X <= 0.15): {p_0_05to0_15:.4f}")

5. Expectation E[X]: 0.1000
6. Variance Var(X):  0.004286
7. P(X <= 0.10):     0.5797
8. P(0.05 <= X <= 0.15): 0.5562


9.

*   The Beta distribution is defined on the interval [0, 1],it's perfectly matches the physical constraints of a proportion or percentage
*   Beta distribution can be skewed to model processes where low defective rates are common.therefore,it is ideal for quality control scenarios.



10.

*   Expectation is 0.1. On average, 10% of the items produced are expected to be defective.


*   Variance is 0.0043. The daily defective rate usually stays very close to the 10% average

